driver/xrt/src/accl.cpp

```diff
@@ -151,6 +151,37 @@ CCLO *ACCL::send(BaseBuffer &srcbuf, unsigned int count,
   return nullptr;
 }
 
+CCLO *ACCL::stream_put(BaseBuffer &srcbuf, unsigned int count,
+                 unsigned int dst, unsigned int stream_id, communicatorId comm_id,
+                 bool from_fpga, streamFlags stream_flags,
+                 dataType compress_dtype, bool run_async,
+                 std::vector<CCLO *> waitfor) {
+  CCLO::Options options{};
+  if (stream_id < 9) return nullptr;
```

Comment:
What is the magic behind the stream IDs? Maybe we could just add 9 to the one provided by the user? Or we could at least throw an exception so the user is aware that this call is not executed.

Reply:
ID's 0-8 will point to internal resources in the CCLO, like various DMA engines, the TX datapath etc. Everything 9+ is directed to the output stream that goes to PL kernels. Agreed on the exception.

Reply:
I would think of stream_id as some kind of tag for streams. Is there a use case where we would use the 0-8 IDs with this call? If not, (and since this offset seems to be fixed) wouldn't it be way cleaner to add this offset directly in the driver (or even in the CCLO) and hide it from the user? With that, we would also have the same behavior for host and HLS API and the user could start from 0, as he/she is used to.

Reply:
There is a use-case, if for example your put would target e.g. the transmit, reduction or compression datapath on the receiver. This isn't something most users would want to do, but it might enable informed users to create lower-latency collective algorithms that bypass memory.